fog/view/server/src/config.rs

```diff
@@ -121,8 +121,14 @@ pub struct FogViewRouterConfig {
     #[clap(long, env = "MC_CLIENT_LISTEN_URI")]
     pub client_listen_uri: RouterClientListenUri,
 
-    /// gRPC listening URI for Fog View Stores.
-    #[clap(long, env = "MC_CLIENT_LISTEN_URI")]
+    /// Sharding Strategies for each Shard. Should be indexed the same as the
+    /// `shard_uris` field.
+    #[clap(long, env = "MC_VIEW_SHARDING_STRATEGIES")]
+    pub sharding_strategies: Vec<ShardingStrategy>,
+
+    /// gRPC listening URI for Fog View Stores. Should be indexed the same as
+    /// the `sharding_stratgies` field.
+    #[clap(long, env = "MC_VIEW_SHARD_URIS")]
     pub shard_uris: Vec<FogViewStoreUri>,
```

Comment:
I'm a little uncomfortable with this, it feels like this maybe should be a URI parameter on the StoreURI so we're not doing this kind of indexing.

e.g. the way we do `?responder-id=some.other.name` could be used to do a `?strategy=epoch`.

Reply:
I agree this is a bit sketchy vis a vis trying to keep these collections in sync. Not sure how clap would handle `Vec<(FogViewStoreUri, ShardingStrategy)>` either. James' suggestion seems reasonable but I haven't really poked around in how `ResponderId` is implemented as a URI parameter so I'm not sure if it'd have any other consequences.

Reply:
I added `FogViewUrl` parsing logic for a query param named `sharding_strategy.` Users can pass `?sharding_strategy=0-100` etc. now.